job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -731,10 +731,13 @@ class PublisherContext extends AbstractExtensibleContext {
      * @since 1.19
      */
     @RequiresPlugin(id = 'groovy-postbuild')
-    void groovyPostBuild(String script, Behavior behavior = Behavior.DoNothing) {
+    void groovyPostBuild(String script, Behavior behavior = Behavior.DoNothing, boolean useSandbox = false) {
         publisherNodes << new NodeBuilder().'org.jvnet.hudson.plugins.groovypostbuild.GroovyPostbuildRecorder' {
-            delegate.groovyScript(script)
-            delegate.behavior(behavior.value)
+            delegate.script {
```

Comment:
This change of the XML structure will break compatibility with older versions of the Groovy Postbuild plugin. Support for older versions of a plugin can only be dropped after a deprecation period. See https://github.com/jenkinsci/job-dsl-plugin/commit/578c0ec5370e8b65bb8fd3f05a43265daf1f2d37 for an example on how to deprecate support for an older plugin version.